moto/kms/models.py

```diff
@@ -714,5 +731,84 @@ def get_public_key(self, key_id: str) -> Tuple[Key, bytes]:
         key = self.describe_key(key_id)
         return key, key.private_key.public_key()
 
+    def rotate_key_on_demand(self, key_id: str) -> str:
+        key: Key = self.keys[self.get_key_id(key_id)]
+
+        rotation = {
+            "KeyId": key_id,
+            "RotationDate": datetime.now().timestamp(),
+            "RotationType": "ON_DEMAND",
+        }
+
+        # Add to key rotations
+        key.rotations.append(rotation)
+
+        return key_id
+
+    def list_key_rotations(
+        self, key_id: str, limit: int, marker: Union[str, None]
+    ) -> Tuple[List[Dict[str, Union[str, float]]], bool, Union[str, None]]:
+        key: Key = self.keys[self.get_key_id(key_id)]
+
+        # we treat the marker as an index-offset
+        marker = marker or "0"
```

Comment:
Does AWS use integers as well for markers, or do they use a random string? If it's random, we can use our '@paginate`-marker - that should simplify the implementation a lot.

See the docs here: https://docs.getmoto.org/en/latest/docs/contributing/development_tips/utilities.html#paginator

Reply:
Somehow missed that this exists, but I should have known that I'm not the first to build a pagination scheme.
I have changed the code to use that utility.